src/test/suite/port-allocation.test.ts

```diff
@@ -0,0 +1,105 @@
+import * as assert from "assert";
+import { TestPortManager } from "../helpers/test-port-manager";
+
+suite("Dynamic Port Allocation Tests", () => {
+  teardown(() => {
+    // Clean up after each test
+    TestPortManager.releasePort();
+  });
+
+  test("allocates available port in test range", async function () {
+    this.timeout(10000);
+
+    const port = await TestPortManager.allocateTestPort();
+    
+    // Verify port is in expected range
+    assert.ok(port >= 9043, `Port ${port} should be >= 9043`);
+    assert.ok(port <= 9999, `Port ${port} should be <= 9999`);
+    
+    // Verify port is not the production port
+    assert.notStrictEqual(port, 9042, "Test port should not be the production port 9042");
+    
+    console.log(`✓ Allocated test port: ${port}`);
+  });
+
+  test("detects production port usage", async function () {
+    this.timeout(5000);
+
+    const productionInUse = await TestPortManager.isProductionPortInUse();
+    console.log(`Production port (9042) in use: ${productionInUse}`);
+    
+    // This test just verifies the detection works - both true/false are valid
+    assert.ok(typeof productionInUse === "boolean", "Should return boolean");
+  });
```

Comment:
This doesn't really verify anything. A better test would be to have an `isPortInUse(portNumber)`, so that a server can intentionally be started and cause a conflict and then see that it returns `true` and when the server shuts down it returns `false`

```suggestion
```